### PR TITLE
Docs: Extend docs with service account limitations

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Extend/update the docs to make it clear that there are endpoints which can't be accessed using service account tokens.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/844

### CODE REVIEW

File: docs/sources/developers/http_api/admin.md [hunk 1]
```diff
@@ -18,9 +18,12 @@ title: 'Admin HTTP API '
...
 
-The Admin HTTP API does not currently work with an API Token. API Tokens are currently only linked to an organization and an organization role. They cannot be given
-the permission of server admin, only users can be given that permission. So in order to use these API calls you will have to use Basic Auth and the Grafana user
-must have the Grafana Admin permission. (The default admin user is called `admin` and has permission to use this API.)
+{{% admonition type="caution" %}}
+The Admin HTTP API currently does not work with service account tokens. Service accounts are limited to an organization and an organization role. They cannot be granted [Grafana server administrator permissions]({{< relref "../roles-and-permissions/#grafana-server-administrators" >}}).
+In order to use these API endpoints you have to use Basic Auth and the Grafana user must have the Grafana server administrator permission.
+
+The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
+{{% /admonition %}}
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/admin.md:26]: - Removes "currently" to make the documentation more timeless: https://developers.google.com/style/timeless-documentation
- Updates the shortcode syntax to match our preferred form which has more consistent semantics (I will open a PR to rewrite all the existing ones later): https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanashortcodes
- Replaces the broken link that uses the `relref` shortcode with a working link that follows our linking guidance: https://grafana.com/docs/writers-toolkit/write/links/
- Modified some language for readability and tone

```suggestion
{{< admonition type="caution" >}}
You can't authenticate to the Admin HTTP API with service account tokens.
Service accounts are limited to an organization and an organization role.
They can't be granted [Grafana server administrator permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#grafana-server-administrators).

To use these API endpoints you have to use Basic authentication and the Grafana user must have the Grafana server administrator permission.

The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
{{< /admonition >}}
```

File: docs/sources/developers/http_api/licensing.md [hunk 2]
```diff
@@ -21,6 +21,13 @@ title: Licensing HTTP API
 
 Licensing is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "/docs/grafana/latest/introduction/grafana-enterprise" >}}).
 
+{{% admonition type="caution" %}}
+The Licensing HTTP API currently does not work with service account tokens. Service accounts are limited to an organization and an organization role. They cannot be granted [Grafana server administrator permissions]({{< relref "../roles-and-permissions/#grafana-server-administrators" >}}).
+In order to use these API endpoints you have to use Basic Auth and the Grafana user must have the Grafana server administrator permission.
+
+The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
+{{% /admonition %}}
+
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/licensing.md:30]: - Removes "currently" to make the documentation more timeless: https://developers.google.com/style/timeless-documentation
- Updates the shortcode syntax to match our preferred form which has more consistent semantics (I will open a PR to rewrite all the existing ones later): https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanashortcodes
- Replaces the broken link that uses the `relref` shortcode with a working link that follows our linking guidance: https://grafana.com/docs/writers-toolkit/write/links/
- Modified some language for readability and tone

```suggestion
{{< admonition type="caution" >}}
You can't authenticate to the Licensing HTTP API with service account tokens.
Service accounts are limited to an organization and an organization role.
They can't be granted [Grafana server administrator permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#grafana-server-administrators).

To use these API endpoints you have to use Basic authentication and the Grafana user must have the Grafana server administrator permission.

The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
{{< /admonition >}}
```

File: docs/sources/developers/http_api/org.md [hunk 3]
```diff
@@ -282,11 +282,12 @@ Content-Type: application/json
...
-only linked to an organization and an organization role. They cannot be given the permission of server
-admin, only users can be given that permission. So in order to use these API calls you will have to
-use Basic Auth and the Grafana user must have the Grafana Admin permission (The default admin user
-is called `admin` and has permission to use this API).
+{{% admonition type="caution" %}}
+The Admin Organizations HTTP API currently does not work with service account tokens. They cannot be granted [Grafana server administrator permissions]({{< relref "../roles-and-permissions/#grafana-server-administrators" >}}).
+In order to use these API endpoints you have to use Basic Auth and the Grafana user must have the Grafana server administrator permission.
+
+The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
+{{% /admonition %}}
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/org.md:290]: - Removes "currently" to make the documentation more timeless: https://developers.google.com/style/timeless-documentation
- Updates the shortcode syntax to match our preferred form which has more consistent semantics (I will open a PR to rewrite all the existing ones later): https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanashortcodes
- Replaces the broken link that uses the `relref` shortcode with a working link that follows our linking guidance: https://grafana.com/docs/writers-toolkit/write/links/
- Modified some language for readability and tone

```suggestion
{{< admonition type="caution" >}}
You can't authenticate to the Admin Organizations HTTP API with service account tokens.
Service accounts are limited to an organization and an organization role.
They can't be granted [Grafana server administrator permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#grafana-server-administrators).

To use these API endpoints you have to use Basic authentication and the Grafana user must have the Grafana server administrator permission.

The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
{{< /admonition >}}
```

File: docs/sources/developers/http_api/user.md [hunk 4]
```diff
@@ -24,11 +24,15 @@ refs:
...
+{{% admonition type="caution" %}}
+The Users HTTP API currently does not work with service account tokens. Service accounts are limited to an organization and an organization role. They cannot be granted [Grafana server administrator permissions]({{< relref "../roles-and-permissions/#grafana-server-administrators" >}}).
+In order to use these API endpoints you have to use Basic Auth and the Grafana user must have the Grafana server administrator permission.
 
-API Tokens can be used with Organization HTTP API to get users of specific organization.
+The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
+
+Service account tokens can be used with Organization HTTP API to get users of specific organization.
+
+{{% /admonition %}}
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/user.md:35]: - Removes "currently" to make the documentation more timeless: https://developers.google.com/style/timeless-documentation
- Updates the shortcode syntax to match our preferred form which has more consistent semantics (I will open a PR to rewrite all the existing ones later): https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanashortcodes
- Replaces the broken link that uses the `relref` shortcode with a working link that follows our linking guidance: https://grafana.com/docs/writers-toolkit/write/links/
- Modified some language for readability and tone

```suggestion
{{< admonition type="caution" >}}
You can't authenticate to the User HTTP API with service account tokens.
Service accounts are limited to an organization and an organization role.
They can't be granted [Grafana server administrator permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#grafana-server-administrators).

To use these API endpoints you have to use Basic authentication and the Grafana user must have the Grafana server administrator permission.

The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
{{< /admonition >}}
```

File: docs/sources/developers/http_api/user.md [hunk 5]
```diff
@@ -24,11 +24,15 @@ refs:
...
-the permission of server users access, only users can be given that permission. To use these API calls you can use Basic Auth and the Grafana
-user must have the Grafana Admin role.
+{{% admonition type="caution" %}}
+The Users HTTP API currently does not work with service account tokens. Service accounts are limited to an organization and an organization role. They cannot be granted [Grafana server administrator permissions]({{< relref "../roles-and-permissions/#grafana-server-administrators" >}}).
+In order to use these API endpoints you have to use Basic Auth and the Grafana user must have the Grafana server administrator permission.
 
-API Tokens can be used with Organization HTTP API to get users of specific organization.
+The `admin` user that Grafana is provisioned with by default has permissions to use these API endpoints.
+
+Service account tokens can be used with Organization HTTP API to get users of specific organization.
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/user.md:33]: Should this be part of the caution in `org.md` instead?

**Review by irenerl24 — APPROVED** [review]
LGTM! Thanks for your comments, @jdbaldry!

File: docs/sources/developers/http_api/user.md (hunk 5 above)
Comment (jdbaldry) [reply docs/sources/developers/http_api/user.md:33]: Oh I see now, that makes sense

Comment (jdbaldry) [reply docs/sources/developers/http_api/user.md:33]: Maybe something like "Alternatively, you can use the Organization HTTP API with service account tokens to get the users for a specific organization"

Comment (IevaVasiljeva) [reply docs/sources/developers/http_api/user.md]: Just changed it to your suggestion, let me know if it looks good to you now. And thanks a lot for making this much neater and providing so much context on your suggestions!

**Review by jdbaldry — APPROVED** [review]
Thank you!

**Pull request merged**